src/events/list/GuildMemberAdd.ts

```diff
@@ -13,9 +15,29 @@ export default class GuildMemberAdd extends Event {
         if (member.user.bot) return;
 
         // Add verification role :
-        const role = await (await Client.instance.getGuild()).roles.fetch(verify.roles.waiting);
+        if (privateMode) {
+            const role = await (await Client.instance.getGuild()).roles.fetch(verify.roles.waiting);
+
+            if (role) member.roles.add(role);
+        } else {
+            const guild = await Client.instance.getGuild();
+            const channel = await guild.channels.fetch(generalChannel);
+
+            if (channel?.type === ChannelType.GuildText) {
+                channel.send({ embeds: [simpleEmbed(msg("event-guildmemberadd-welcome"), "normal")] });
+            }
+
+            const tier = await gqlRequest<GetMemberActivityTierType, GetMemberActivityTierVariables>(getMemberActivityTier, {
+                memberId: member.id
+            });
+
+            if (!tier.data?.member.activity.tier) return;
```

Comment:
Il ne faut pas return ici, car l'exécution du code n'est pas fini en dessous ("Create the member if he dosen't exist"), de plus généralement si le membre n'a pas de tier, ça veux justement dire qu'il n'existe pas encore dans la BDD du Royaume donc il doit être créé.